src/online-members/script.js

```diff
@@ -162,13 +162,16 @@ async function generateMembersList() {
 }
 
 function addEventListenerToMembersList() {
-  const membersList = document.querySelector(`#${MEMBERS_CONTAINER_ID} > ul`);
+  const membersList = document.querySelector(`#${MEMBERS_CONTAINER_ID} > UL`);
   membersList.addEventListener('click', (event) => {
-    const membersDiv = event.target.nodeName === 'DIV';
-    if (!membersDiv) {
+    const memberElement = event.target.closest(`.${MEMBERS_CLASS}`);
+    if (!memberElement) {
       return;
     }
-    const username = event.target.dataset.username;
+    const username = memberElement.dataset.username;
```

Comment:
Should we add null checks here for dataset ?

Reply:
Isn't that being handled in the next line? or am I missing something here?

Reply:
I am wondering if it's possible for dataset to be null/undefined, in that case it would be something like memeberElement.undefined.username which would throw an error
Please correct me if I am worng in understanding this

Reply:
Dataset won't be null, it is a property that is available for all elements in HTML, so if set will return the value, or else will return undefined if values are not set, Please let me know if I am missing something

Reply:
Got it